### PR TITLE
testserver: ensure no whitespace in URLs

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach-go/34)

### CODE REVIEW

File: testserver/testserver_test.go [hunk 1]
```diff
@@ -26,8 +27,21 @@ func TestRunServer(t *testing.T) {
...
+
+func TestPGURLWhitespace(t *testing.T) {
+	ts, err := testserver.NewTestServer()
 	if err != nil {
 		t.Fatal(err)
 	}
+	if err := ts.Start(); err != nil {
+		t.Fatal(err)
+	}
+	url := ts.PGURL().String()
```
Comment (jordanlewis) [inline testserver/testserver_test.go]: when would this return a url with a space in the first place?

File: testserver/testserver.go [hunk 2]
```diff
@@ -263,7 +264,7 @@ func (ts *TestServer) pollListeningURLFile() error {
 		time.Sleep(100 * time.Millisecond)
 	}
 
-	u, err := url.Parse(string(data))
+	u, err := url.Parse(string(bytes.TrimSpace(data)))
```
Comment (jordanlewis) [inline testserver/testserver.go]: should we urlencode the spaces?

File: testserver/testserver_test.go (hunk 1 above)
Comment (tamird) [inline testserver/testserver_test.go]: Yep! The test fails before this change.

File: testserver/testserver.go (hunk 2 above)
Comment (tamird) [inline testserver/testserver.go]: In this case we're just seeing the trailing newline in the url file. If you can help me conceive of a test that would exercise legitimate spaces in the URL I'd be happy to adding urlencoding.

**Comment by tamird** [comment]
Discussed in person, merging.

**Pull request merged**